src/main/java/com/redhat/cloud/policies/app/model/history/PoliciesHistoryRepository.java

```diff
@@ -84,12 +84,14 @@ public List<PoliciesHistoryEntry> find(String tenantId, UUID policyId, Pager pag
 
     private static String addFiltersConditions(String hql, List<Filter.FilterItem> filterItems) {
         // The filters from the pager are added to the HQL query.
+        StringBuilder hqlBuilder = new StringBuilder(hql);
         for (Filter.FilterItem filterItem : filterItems) {
             String entityFieldName = getEntityFieldName(filterItem);
             String operator = getHqlOperator(filterItem);
             // To be consistent with the previous implementation, the condition is always case-insensitive.
-            hql += " AND LOWER(" + entityFieldName + ")" + operator + ":" + entityFieldName;
+            hqlBuilder.append(" AND LOWER(").append(entityFieldName).append(")").append(operator).append(":").append(entityFieldName);
```

Comment:
`StringBuilder` is slower in modern Java than the `+` operator when the number of iterations is reasonable (it is here). The opposite was true in the past.